test/functional/lua/buffer_updates_spec.lua

```diff
@@ -216,4 +216,23 @@ describe('lua: buffer event callbacks', function()
     eq(1, meths.get_var('listener_cursor_line'))
   end)
 
+  it('does not SEGFAULT when calling win_findbuf in on_detach', function()
+
+    exec_lua[[
+      local buf = vim.api.nvim_create_buf(false, false)
+
+      vim.cmd"split"
+      vim.api.nvim_win_set_buf(0, buf)
+
+      vim.api.nvim_buf_attach(buf, false, {
+        on_detach = function(_, buf)
+          vim.fn.win_findbuf(buf)
+        end
+      })
+    ]]
+
+    feed('ZQ')
```

Comment:
nit: is there a reason not to use `command('quit!')` here? The benefit of `command()` is that it's synchronouse, whereas feed() just sends input to the nvim server and doesn't (can't) wait for nvim to process it.

Reply:
Updated to use `command`  but with `q!` as `qa!` does not trigger the segfault in master while `q!` does.